drivers/serial/uart_stm32.c

```diff
@@ -1939,12 +1951,12 @@ static const struct uart_stm32_config uart_stm32_cfg_##index = {	\
 	.de_assert_time = DT_INST_PROP(index, de_assert_time),		\
 	.de_deassert_time = DT_INST_PROP(index, de_deassert_time),	\
 	.de_invert = DT_INST_PROP(index, de_invert),			\
+	.uart_cfg = &uart_cfg_##index,					\
 	STM32_UART_IRQ_HANDLER_FUNC(index)				\
 	STM32_UART_PM_WAKEUP(index)					\
 };									\
 									\
 static struct uart_stm32_data uart_stm32_data_##index = {		\
-	.baud_rate = DT_INST_PROP(index, current_speed),		\
 	.reset = RESET_DT_SPEC_GET(DT_DRV_INST(index)),			\
```

Comment:
I could move this over while I'm already making changes to config/data.

Reply:
Would be great, txs

Reply:
The way the DMA stuff is written, it [modifies its properties](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_stm32.c#L1156-L1209) during runtime. So this would require a significant rework that should probably be a different PR.